xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java

```diff
@@ -132,28 +132,38 @@
     @Override
     public ResourceReference parse(String rawReference)
     {
-        // Step 1: Check if it's a known URI by looking for one of the known URI schemes. If not, check if it's a URL.
-        ResourceReference resourceReference = parseURILinks(rawReference);
-        if (resourceReference != null) {
-            return resourceReference;
-        }
+        while(true) {
```

Comment:
I really don't like the use of while for something we never intended to repeat.

Reply:
Wait I forgot to add a break statement at the end.

Reply:
Ok done.

Reply:
It's still a unreadable way to use while. Since the whole point here is to improve codestyle it's really not a good idea...

Reply:
🤔 I thought it's because multiple returns are hard to debug.

Reply:
Multiple return are not very hard to debug. It's just that it's a bit harder to know where are the exit points of the method. But you are not really improving that here.